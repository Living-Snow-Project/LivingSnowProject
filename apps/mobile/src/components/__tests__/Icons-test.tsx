import React from "react";
import { Platform } from "react-native";
import { render } from "@testing-library/react-native";
import {
  DeleteIcon,
  EditIcon,
  PictureIcon,
  RecordIcon,
  ScrollTopIcon,
  SnowIcon,
} from "../Icons";

describe("Icons test suite", () => {
  describe("android", () => {
    beforeEach(() => {
      Platform.OS = "android";
    });

    afterEach(() => {
      Platform.OS = "ios";
    });

    test("sample record icon", () => {
      const { toJSON } = render(<RecordIcon type="Sample" />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("sighting record icon", () => {
      const { toJSON } = render(<RecordIcon type="Sighting" />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("picture icon", () => {
      const { toJSON } = render(<PictureIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("scroll top icon", () => {
      const { toJSON } = render(<ScrollTopIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("snow icon", () => {
      const { toJSON } = render(<SnowIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("edit icon", () => {
      const { toJSON } = render(<EditIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("delete icon", () => {
      const { toJSON } = render(<DeleteIcon />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe("ios", () => {
    test("sample record icon", () => {
      const { toJSON } = render(<RecordIcon type="Sample" />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("sighting record icon", () => {
      const { toJSON } = render(<RecordIcon type="Sighting" />);
      expect(toJSON()).toMatchSnapshot();
      Platform.OS = "ios";
    });

    test("picture icon", () => {
      const { toJSON } = render(<PictureIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("scroll top icon", () => {
      const { toJSON } = render(<ScrollTopIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("snow icon", () => {
      const { toJSON } = render(<SnowIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("edit icon", () => {
      const { toJSON } = render(<EditIcon />);
      expect(toJSON()).toMatchSnapshot();
    });

    test("delete icon", () => {
      const { toJSON } = render(<DeleteIcon />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
