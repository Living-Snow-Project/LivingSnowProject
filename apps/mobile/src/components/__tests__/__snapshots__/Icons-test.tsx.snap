// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons test suite android delete icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "darkred",
        "fontSize": 32,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="delete-icon"
>
  
</Text>
`;

exports[`Icons test suite android edit icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "green",
        "fontSize": 32,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="edit-icon"
>
  
</Text>
`;

exports[`Icons test suite android picture icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "green",
        "fontSize": 48,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="picture-icon"
>
  
</Text>
`;

exports[`Icons test suite android sample record icon 1`] = `<Text />`;

exports[`Icons test suite android scroll top icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontSize": 48,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="scroll-top-icon"
>
  
</Text>
`;

exports[`Icons test suite android sighting record icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "skyblue",
        "fontSize": 48,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="sighting-icon"
>
  
</Text>
`;

exports[`Icons test suite android snow icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#ccc",
        "fontSize": 32,
      },
      Object {},
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="snow-icon"
>
  
</Text>
`;

exports[`Icons test suite ios delete icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "darkred",
        "fontSize": 32,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="delete-icon"
>
  
</Text>
`;

exports[`Icons test suite ios edit icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "green",
        "fontSize": 32,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="edit-icon"
>
  
</Text>
`;

exports[`Icons test suite ios picture icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "green",
        "fontSize": 48,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="picture-icon"
>
  
</Text>
`;

exports[`Icons test suite ios sample record icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "lightcoral",
        "fontSize": 48,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="sample-icon"
>
  
</Text>
`;

exports[`Icons test suite ios scroll top icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontSize": 48,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="scroll-top-icon"
>
  
</Text>
`;

exports[`Icons test suite ios sighting record icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "skyblue",
        "fontSize": 48,
      },
      undefined,
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="sighting-icon"
>
  
</Text>
`;

exports[`Icons test suite ios snow icon 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#ccc",
        "fontSize": 32,
      },
      Object {},
      Object {
        "fontFamily": "ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
  testID="snow-icon"
>
  
</Text>
`;
